cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/SourceControlManagementService.java

```diff
@@ -223,11 +224,11 @@ public ApplicationRecord pullAndDeploy(ApplicationReference appRef) throws Excep
    * @return {@link PullAppResponse}
    * @throws NoChangesToPullException if the fileHashes are the same
    * @throws NotFoundException if the repository config is not found or the application in repository is not found
-   * @throws PullFailureException if unexpected errors happen when pulling the application.
+   * @throws SourceControlException if unexpected errors happen when pulling the application.
    * @throws AuthenticationConfigException if the repository configuration authentication fails
    */
   private PullAppResponse<?> pullAndValidateApplication(ApplicationReference appRef)
-    throws NoChangesToPullException, NotFoundException, PullFailureException, AuthenticationConfigException {
```

Comment:
Is it because `SourceControlException` i's runtime exception we omit it in the throws list?